### ISSUE TITLE
Classbalancing pipeline operator when split data

### ISSUE THREAD

**Issue by invain1218** [mlr-org/mlr3pipelines#842]
Hello, I’m a bit confused about the `po("classbalancing") `pipeline. If I add po("classbalancing") to my model, for instance,`learner = po("classbalancing") %>>% lrn("ranger")`, and then split my data, will this learner apply class balancing to both the training and test sets?

I only want it to balance the training data, so should I do something like `pro_train_task = po("classbalancing")$train(task, split$train)`? I’ve understood that pipelines added directly to the learner would also apply to the test set, but I’m not entirely sure. I would really appreciate some clarification on this. Thank you so much!

**Comment by mb706** [comment]
The classbalancing pipeop only balances training data and does nothing to test data / data during inference. As it says [in the docs](https://mlr3pipelines.mlr-org.com/reference/mlr_pipeops_classbalancing.html?q=Sampling+happens+only+during+training+phase):

> Sampling happens only during training phase.

More generally, mlr3 requires that there should always be exactly one prediction made for each row in the prediction task, so PipeOps that change the number of rows only ever do that during training, not during prediction.

**Comment by mb706** [comment]
The way you call the objects should be exactly equivalent. It is possible that the construction of objects somehow uses the RNG and therefore changes results. If you set the seed after having constructed all your objects, it should work the same. I.e. if you do
```r
# construct objects
pos = cate_po %>>% num_po
lr = lrn(“classif.ranger”)

# run
set.seed(1)
res = pos$train(task)[[1]]
res2 = pob$train(list(res))[[1]]
model = lr$train(res2)
model$predict(res2)$score(msr(“classif.acc”))
```
the result should be the same as if when you do
```r
# construct objects
pos = cate_po %>>% num_po %>>% pob
grah = as_learner(pos %>>% lrn(“classif.ranger”))

# run
set.seed(1)
model = grah$train(task)
model$predict(task)$score(msr(“classif.acc”))
```

**Comment by invain1218** [comment]
Thanks vary much!!! I’ve got it now.
